backend/src/junior/pipes/phoneNumberValidation.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import { PipeTransform, BadRequestException } from "@nestjs/common";
+import * as content from '../../content.json';
+
+// Custom pipe for handling "phoneNumber" and "parentsPhoneNumber" validation while adding/editing junior details
+export class PhoneNumberValidationPipe implements PipeTransform {
+  readonly allowedPhoneNumber = /(^(\+358|0|358)\d{9})/;
```

Comment:
readonly allowedPhoneNumber = /(^(\+358|0|358)\d{9})/; -> better be /(^(\+358|0|358)\d{9}$)/, otherwise a longer phone number can be accepted too

Reply:
Thanks and good catch. Will fix it right away

Reply:
@mariiu 
Got this error with the regular expression:
```
SyntaxError: Invalid regular expression: /(^(+358|0|358)\d{9}$)/: Nothing to repeat
    at new PhoneNumberValidationPipe (/Users/vinu.joseph/Gofore/City-of-Vantaa-SmartLab/Mobiilinutakortti/backend/dist/junior/pipes/phoneNumberValidation.pipe.js:7:9)
    at Injector.instantiateClass (/Users/vinu.joseph/Gofore/City-of-Vantaa-SmartLab/Mobiilinutakortti/backend/node_modules/@nestjs/core/injector/injector.js:288:19)
    at callback (/Users/vinu.joseph/Gofore/City-of-Vantaa-SmartLab/Mobiilinutakortti/backend/node_modules/@nestjs/core/injector/injector.js:75:41)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```